src/argilla/client/feedback/dataset/local.py

```diff
@@ -0,0 +1,113 @@
+from argilla.client.feedback.types import AllowedFieldTypes, AllowedQuestionTypes
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.schemas import FeedbackRecord
+
+
+warnings.simplefilter("always", DeprecationWarning)
+
+
+class FeedbackDataset(FeedbackDatasetBase, ArgillaToFromMixin):
+    def __init__(
+        self,
+        *,
+        fields: List[AllowedFieldTypes],
+        questions: List[AllowedQuestionTypes],
+        guidelines: Optional[str] = None,
+    ) -> None:
+        super().__init__(fields=fields, questions=questions, guidelines=guidelines)
+
+    def __len__(self) -> int:
+        """Returns the number of records in the dataset."""
+        return len(self.records)
+
+    def __getitem__(self, key: Union[slice, int]) -> Union["FeedbackRecord", List["FeedbackRecord"]]:
+        """Returns the record(s) at the given index(es).
+
+        Args:
+            key: the index(es) of the record(s) to return. Can either be a single index or a slice.
+
+        Returns:
+            Either the record of the given index, or a list with the records at the given indexes.
+        """
+        if len(self.records) < 1:
+            raise RuntimeError(
+                "In order to get items from `FeedbackDataset` you need to add them first" " with `add_records`."
+            )
+        if isinstance(key, int) and len(self.records) < key:
+            raise IndexError(f"This dataset contains {len(self)} records, so index {key} is out of range.")
+        return self.records[key]
+
+    def iter(self, batch_size: Optional[int] = FETCHING_BATCH_SIZE) -> Iterator[List["FeedbackRecord"]]:
+        """Returns an iterator over the records in the dataset.
+
+        Args:
+            batch_size: the size of the batches to return. Defaults to 100.
+        """
+        for i in range(0, len(self.records), batch_size):
+            yield self.records[i : i + batch_size]
+
+    def fetch_records(self) -> None:
+        warnings.warn(
+            "As the current `FeedbackDataset` is stored locally and not pushed to Argilla,"
+            " the method `fetch_records` won't do anything. If you want to fetch the records"
+            " from Argilla, make sure you're using an `RemoteFeedbackDataset`, either by"
+            " calling `FeedbackDataset.from_argilla` or by keeping the returned value from"
+            " `FeedbackDataset.push_to_argilla`.",
+            DeprecationWarning,
+        )
+
+    @property
+    def argilla_id(self) -> None:
+        warnings.warn(
+            "As the current `FeedbackDataset` is stored locally, `argilla_id` won't"
+            " return anything as it's not pushed to Argilla. If you want to get the id"
+            " of a dataset in Argilla, make sure you're using an `RemoteFeedbackDataset`,"
+            " either by calling `FeedbackDataset.from_argilla` or by keeping the returned"
+            " value from `FeedbackDataset.push_to_argilla`.",
+        )
+
+    def add_records(
```

Comment:
A question here. For local datasets, we could do something like:
```python
for data in ...:
   dataset.records.append(FeedbackRecord(**data))
``` 
to add records, right? I mean, it's possible right now, right? Maybe we should define the records as a property based on a protected attribute `__records`

Reply:
Sure, I can make `records` a property